### PR TITLE
fix(curriculum): Updated step 74 of spreadsheet project and added step 75 and step 76

### LINKED ISSUE #54327: The system raises an error even for correct answers, and does not allow you to proceed to the next challenge
### Describe the Issue

My solution meets all the challenge requests, but is rejected with incomprehensible exceptions.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-functional-programming-by-building-a-spreadsheet/step-74

### Your code

```
const highPrecedence = str => {
  const regex = /^(\d*\.?\d+)\s*([\*\/])\s*(\d*\.?\d+)$/;
}
```

### Expected behavior

My solution should pass the challenge.

### Screenshots

_No response_

### System

- Device: Laptop
- OS: MacOs Ventura
- Browser: Chrome
- Version: 123.0.6312.107

### Additional context

Here (https://forum.freecodecamp.org/t/learn-functional-programming-by-building-a-spreadsheet-step-74/684317/3?u=sergio.am.spina) the correctness of my solution was recognized (and perhaps the incorrectness of the official solution), and here (https://forum.freecodecamp.org/t/learn-functional-programming-by-building-a-spreadsheet-step-74/684317/5?u=sergio.am.spina) it was suggested to me to open an issue.

### PR DESCRIPTION
**Pull request by sanskriti2005**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54327

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d404259f512c1a9e86ac1.md [hunk 1]
```diff
@@ -11,72 +11,48 @@ In your `highPrecedence` function, declare a `regex` variable. Assign it a regul
...
 
-Your `regex` should use a second capture group.
+Your highPrecedence should return `regex.test(str);`
 
 ```js
-assert.match(code, /const\s+highPrecedence\s*=\s*(\(\s*str\s*\)|str)\s*=>\s*{\s*const\s+regex\s*=\s*\/\(\[(?:\\d\.|\.\\d)\]\+\)\(/);
-```
-
-Your second capture group should match a `*` or `/` operator. Use a character class in the capture group.
+assert.match(code, /return\s+regex\.test\(str\);/);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d404259f512c1a9e86ac1.md:46]:
```suggestion
assert.match(code, /return\s+regex\.test\(str\);?/);
```

Optional semi.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+---
+id: 661f48f412d7631a1d9c30e6
+title: Step 75
+challengeType: 0
+dashedName: step-75
+---
+
+# --description--
+
+You should use `console.log()` to return the `highPrecedence` function with the string `"5*3"`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:10]:
```suggestion
You should use `console.log()` to print the result of calling the `highPrecedence` function with the string `"5*3"`.
```

`console.log()` returns `void`.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+# --hints--
+
+Please return the function using `console.log(highPrecedence("5*3"));`.
+
+```js
+
+assert.match(code, /console\.log\(highPrecedence\("5\*3"\)\);/);
+
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:20]: We should make this more granular.

- Test for a function call of `highPrecedence`
- Test that `5*3` is passed as the argument
- Test that the result is logged.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
+---
+id: 661f49650572031c6ebdb8e3
+title: Step 76
+challengeType: 0
+dashedName: step-76
+---
+
+# --description--
+
+Remove line code line `return regex.test(str);` and line `console.log(highPrecedence("5*3"))` as they were only for testing purposes.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md:10]:
```suggestion
Remove both the `console.log()` with your `highPrecedence` call, and the `return` statement from your `highPrecedence` function.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+# --hints--
+
+Remove the line `console.log(highPrecedence("5*3"))`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md:14]:
```suggestion
Your code should not log the result of `highPrecedence("5*3")`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+```
+
+Remove the line `return regex.test(str);`.
+
+```js
+
+assert.notMatch(code, /return\s+regex\.test\(str\);/);
+
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md:28]:
````suggestion
Your `highPrecedence` function should not return a value.

```js
assert.isUndefined(highPrecedence("5*3"));
```
````

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+# --hints--
+
+Remove the line `console.log(highPrecedence("5*3"))`.
+
+```js
+
+assert.notMatch(code, /console\.log\(highPrecedence\("5\*3"\)\)/);
+
+```
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f49650572031c6ebdb8e3.md:20]:
````suggestion
```js
assert.notMatch(code, /console\.log\(highPrecedence\("5\*3"\)\)/);
```
````

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+dashedName: step-75
+---
+
+# --description--
+
+You should use `console.log()` to print the result of calling the `highPrecedence` function with the string `"5*3"`.
+
+# --hints--
+
+Print or log the results using `console.log()`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:14]:
```suggestion
You should call `console.log()`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+```js
+
+assert.match(code, /console\.log\(.*\);/);
+
+```
+
+Make a test call for the `highPrecedence` function.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:22]:
````suggestion
You should call your `highPrecedence` function.

```js
assert.match(code, /highPrecedence\(.*\)/);
```

You should pass your `highPrecedence` call to your `console.log()` call.
````

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:1]: The sei-colons here should be optional

Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:1]: Also it's possible for the regex tests to not be this complicated

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+You should use `console.log()` to print the result of calling the `highPrecedence` function with the string `"5*3"`.
+
+# --hints--
+
+You should call `console.log()`.
+
+```js
+
+assert.match(code, /console\.log\(.*\);/);
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:18]: This works fine: 
```suggestion
assert.match(code, /console\.log\(/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+
+```
+
+You should call your `highPrecedence` function.
+
+```js
+
+assert.match(code, /console\.log\(highPrecedence\((?:"[^"]*"|[^)]*)\)\);/);
+
+
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:28]: This works fine too:
```suggestion
assert.match(code, /console\.log\(highPrecedence\(/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+Pass `5*3` as the argument
+
+```js
+
+assert.match(code, /console\.log\(highPrecedence\("5\*3"\)\);/);
+
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md:36]: And this works fine:
```suggestion
assert.match(code, /console\.log\(highPrecedence\(["']5\*3["']\)\);?/);
```

We should allow single quotes, and make the semi-colon optional.

**Comment by sanskriti2005** [comment]
@Ksound22 thank you for the review :)  

I will make the changes.

**Review by Ksound22 — COMMENTED** [review]
LGTM 👍🏾

**Review by Ksound22 — APPROVED** [review]

**Pull request merged**